pointmatcher/DataPointsFilters/NormalSpace.cpp

```diff
@@ -176,10 +176,13 @@ template <typename T>
 std::size_t NormalSpaceDataPointsFilter<T>::bucketIdx(T theta, T phi) const
 {
 	//Theta = polar angle in [0 ; pi] and Phi = azimuthal angle in [0 ; 2pi]
-	assert((theta >= 0.0) && (theta <= static_cast<T>(M_PI)) && (phi >= 0) && (phi <= 2.0*((float) static_cast<T>(M_PI))));
+	assert( (theta >= 0.0) && (theta <= static_cast<T>(M_PI)) && "Theta not in [0, Pi]");
+	assert( (phi >= 0) && (phi <= 2.0*((float) static_cast<T>(M_PI))) && "Phi not in [0, 2Pi]");
```

Comment:
I don't fully understand why they had the cast to float.

Reply:
Oops. I think this was an artifact of when I was debugging the hashing function and should be removed.

The static_cast<T> is intentional to prevent weird behavior with mixed single and double precision math, since the values are directly used as part of the hashing function. M_PI is just a macro for 3.14159..., and I wanted to make sure that the values were being cast to the correct datatype to check for bounds. The values are not cast as part of the hashing function because the asserts should handle hashing errors at test time.  It's also due to a lack of my own familiarity of exactly when the compiler decides to use single and double precision math or might mix them.